hazelcast/src/main/resources/release_notes.txt

```diff
@@ -1,384 +1,18 @@
-* The `recreateCachesOnCluster` invocation is not being checked
-for the maximum invocations count anymore during cluster restarts. [#16014]
-
-2. Fixes
-
-* Fixed an issue where repetitive calls of `IMap.loadAll()`
-may cause memory leaks. [#16171]
-* Fixed an issue where an HTTP request via
-`hazelcast/rest/cluster` was failing when using the advanced network
-configuration and a client endpoint is
-not specified. In this case, this request's response
-has been improved to report `0` as the client connection count. [#16150]
-* Fixed an issue where an enabled `redoOperation()` was not
-throwing an exception when an empty list is tried to be retrieved
-on the client. [#16019]
-* Fixed an issue where gathering MultiMap statistics was breaking
-the split-brain healing with `LatestAccessMergePolicy`. [#16003]
-* Fixed the `multiplier` element of client's connection
-retry configuration to be set as a `double` in the
-XML configuration (`hazelcast-client.xml). [#15990]
-* Fixed the following Hot Restart issues:
-   ** Introduced an additional stage to the Hot Restart
-procedure, i.e., `HotRestartIntegrationService.startup()`,
-which waits until all members transition from the `PASSIVE` state.
-This guarantees all members to have the same state after Hot
-Restart operation is finished.
-   ** IMap proxies created during Hot Restart are not initialized
-and published to other cluster members. So the operation
-has been improved to force initialize any uninitialized proxies
-and publish them. This fixed the issue where the
-`getDistributedObjects()` method was not reporting the persisted
-objects after a Hot Restart.
-[#15986]
-* Fixed an issue where, after forming a cluster, certain members
-occasionally do not contain `EventService` registrations for
-some other members. [#15950]
-* Fixed an issue where calling the `IMap.removeAll()` method
-without index was updating the last access and expiry time
-for all records. [#15853]
-* Fixed an issue where the client connection count
-was retrieved using an incorrect method. [#15681]
-
-==== 3.12.4 ====
-
-1. Enhancements
-
-- Added support for the `PartitionLostListener` configuration
-on the client side. [#15762]
-
-2. Fixes
-
-- Fixed an issue where the query was ignoring the index
-but was processing successfully. [#15808]
-- Fixed an issue where the predicate queries to retrieve
-multi-values were failing on the nested JSON objects in arrays. [#15789]
-- Fixed a possible livelock when many threads were trying to
-use the `incrementMod()` method causing an infinite loop. [#15777]
-- Fixed shutting down Management Center service caused by
-an NPE when MultiMap statistics are gathered after
-`MultiMapService.shutdown()`. [#15656]
-
-==== 3.12.3 ====
+This document lists the new features, enhancements, fixed issues and, removed or deprecated features for Hazelcast IMDG 4.0.2-Migration release. The numbers in the square brackets refer to the issues in Hazelcast's GitHub repositories.
```

Comment:
maybe let's change the release numbering there.. IMDG 3.12.8-Migration ...

Reply:
Done, thanks!